src/System.CommandLine.Tests/ApprovalTests/Help/HelpBuilderTests.Approval.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+using System.CommandLine.Help;
+using System.IO;
+using ApprovalTests;
+
+
+namespace System.CommandLine.Tests.Help
+{
+    public partial class HelpBuilderTests
+    {
+        [Fact]
+        public void Help_describes_default_values_for_complex_root_command_scenario()
+        {
+            var command = new RootCommand(description: "Test description")
+            {
+                new Argument<string>("the-root-arg-no-default")
+                {
+                    Description = "the-root-arg-no-default-description",
+                },
+                new Argument<string>("the-root-arg", () => "the-root-arg-one-value")
+                {
+                    Description = "the-root-arg-description"
+                },
+                new Argument<FileAccess>("the-root-arg-enum-default", () => FileAccess.Read)
+                {
+                    Description = "the-root-arg-enum-default-description",
+                    ArgumentType = typeof(FileAccess)
+                },
+                new Option(aliases: new string[] {"--the-root-option-no-arg", "-trna"}) {
+                    Description = "the-root-option-no-arg-description",
+                    Required = true
+                },
+                new Option(aliases: new string[] {"--the-root-option-no-default-arg", "-tronda"}) {
+                    Description = "the-root-option-no-default-description",
+                    Argument = new Argument<string>("the-root-option-arg-no-default-arg")
```

Comment:
For these we should prefer `Option<T>` over explicitly setting the `Argument` property.

Reply:
I've found a suitable way to create an option that takes an argument with description. I will add the case you mentioned above as a separate option example. (trondda)

```csharp
new Option<string>(
    aliases: new string[] {"--the-root-option-no-description-default-arg", "-trondda"}, 
    parseArgument: _ => "the-root-option--no-description-default-arg-value",
    isDefault: true
),
```